apps/website/src/middleware.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  type NextFetchEvent,
+  type NextRequest,
+  NextResponse,
+} from "next/server";
+import { env } from "@/env/index.mjs";
+import { callEndpoint } from "@/server/utils/queue";
+import type { TrackClickSchema } from "@/pages/api/short-links/track-click";
+
+export async function middleware(req: NextRequest, event: NextFetchEvent) {
+  const res = await fetch(env.NEXT_PUBLIC_BASE_URL + "/api/short-links", {
+    method: "POST",
+    body: JSON.stringify({
+      slug: req.nextUrl.pathname.replace("/l/", "").trim(),
+    }),
+  });
+  const shortLink = await res.json();
+
+  if (shortLink) {
+    event.waitUntil(
+      callEndpoint<TrackClickSchema>("/api/short-links/track-click", {
+        id: shortLink.id,
+      }),
+    );
+    return NextResponse.redirect(shortLink.link);
```

Comment:
Should we add the status code 308 or is 307 fine?

Reply:
I'll leave that up to @MattIPv4. Im happy with the temporary redirect as the target can change. I think we discussed this in regards to the next config redirects.

Reply:
Temporary is preferred here I think 👍 Anything permanent should probably be in the next config anyway rather than using this system